src/Core/Commit.js

```diff
@@ -0,0 +1,314 @@
+				return createElement(element)
+			case Boolean:
+			case Symbol:
+				break
+			default:
+				return elementUnknown(element)
+		}
+
+	return elementText('')
+}
+
+/**
+ * @param {Element} element
+ * @param {Element} snapshot
+ */
+function commitPromise (element, snapshot) {
+	snapshot.type.then(function (value) {
+		if (!element.DOM)
+			return
+		if (element.flag === ElementPromise)
+			reconcileChildren(element, elementFragment(commitElement(value)))
+		else
+			reconcileElement(element, commitElement(value))
+	})
+}
+
+/**
+ * @param {Element} element
+ * @param {Element} sibling
+ * @param {Element} host
+ */
+function commitChildren (element, sibling, host) {
+	var children = element.children
+	var length = children.length
+	var next = children.next
+
+	while (length-- > 0) {
+		commitMount(!next.DOM ? next : merge(new Element(ElementNode), next), sibling, element, host, 0)
+		next = next.next
+	}
+}
+
+/**
+ * @param {Element} element
+ * @param {Element} sibling
+ * @param {Element} parent
+ * @param {Element} host
+ * @param {number} signature
+ */
+function commitMount (element, sibling, parent, host, signature) {
+	element.host = host
+	element.parent = parent
+	element.context = host.context
+
+ 	switch (element.flag) {
+ 		case ElementComponent:
+ 			element.work = WorkTask
+ 			
+ 			componentMount(element)
+
+ 			if (element.owner[LifecycleWillMount]) 
+ 				lifecycleMount(element, LifecycleWillMount)
+
+ 			commitMount(element.children, sibling, parent, element, signature)
+
+ 			if ((element.DOM = element.children.DOM, element.ref)) 
+ 				commitRef(element, element.ref, 1)
+ 			
+ 			if (element.owner[LifecycleDidMount])
+ 				lifecycleMount(element, LifecycleDidMount)
+
+ 			element.work = WorkSync
+ 			return
+ 		case ElementPortal:
+ 			element.DOM = DOM(element.type)
+ 			break
+ 		case ElementPromise:
+ 			commitPromise(element, element)
+ 		case ElementFragment:
+ 			element.DOM = DOM(DOMNode(parent))
+ 			break
+ 		case ElementNode:
+ 			element.xmlns = DOMScope(element.type, parent.xmlns)
+ 		case ElementText:
+ 			element.DOM = commitDOM(element)
+ 			
+ 			if (signature < 1) 
+ 				commitAppend(element, parent)
+ 			else
+ 				commitInsert(element, sibling, parent)
+ 
+ 			if (element.flag > ElementNode)
+ 				return
+ 	}
+
+	commitChildren(element, element, host)
+	commitProps(element, element.props, 1)
+}
+
+/**
+ * @param {Element} element
+ * @param {Element} parent
+ * @param {number} signature
+ */
+function commitUnmount (element, parent, signature) {
+	if (element.flag === ElementComponent)
+		return componentUnmount(element, element.children, parent, signature, 1)
+
+	commitRemove(element, parent)
+	commitDetach(element, signature)
+}
+
+/**
+ * @param {Element} element
+ * @param {Element} snapshot
+ * @param {number} signature
+ */
+function commitReplace (element, snapshot, signature) {
+	if (signature > 0 && commitUnmount(element, element.parent, 1))
+		return void element.state.then(function () {
```

Comment:
missing promise error handling
cool use of `void`, I wonder if performance is equivalent to empty return on new line